pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+  - EnumerateInstances - Gets instances for a class defined in a namespace
+    in the repository.  If there is no class repository, it gets instances
+    for the defined class.  If a class definition for the target class is in
+    the repository, it attempts to get instances of this class and all
+    subclasses
+  - EnumerateInstanceNames - Gets instance names for a class defined in
+    a namespace in the repository.  If there is no class repository, it gets
+    names or the defined class.  If a class definition for the target class is
+    in the repository, it attempts to get instances of this class and all
+    subclasses
+  - CreateInstance - TODO
+  - ModifyInstance - TODO
+  - ExecQuery - This method will return fixed data based on the query string.
+    It does not execute the select statement itself nor does it parse it.
+    An incorrect select statement will be ignored.
+
+- Associators and References
+ - AssociatorNames
+ - Associators
+ - ReferenceNames
+ - References
+
+Supports both class and instance methods
+
+-InvokeMethod
+
+- Pull Operaions - All of the Open and Pull methods are included. However,
+  any Query Language statements are ignored and the full instance responses
+  generated. Since FQL only filters the set of instanced defined by
+  the corresponding request (i.e. Open/Pull EnumerateInstances simply
+  filters instances that would be acquired by EnumerateInstances. This means
+  that these methods can be used but may return more instances than would
+  be returned without the filters.
+
+These methods require the required classes and instances to be in the repository
+for what is expected to be returned
+"""
+from __future__ import absolute_import, print_function
+
+import copy
+import uuid
+import time
+from mock import Mock
+import six
+
+from pywbem import WBEMConnection, CIMClass, CIMClassName, \
+    CIMInstance, CIMInstanceName, CIMQualifierDeclaration, \
+    CIMError, \
+    CIM_ERR_NOT_FOUND, CIM_ERR_FAILED, CIM_ERR_INVALID_SUPERCLASS, \
+    CIM_ERR_INVALID_PARAMETER, CIM_ERR_INVALID_CLASS, CIM_ERR_ALREADY_EXISTS, \
+    CIM_ERR_INVALID_NAMESPACE, CIM_ERR_INVALID_ENUMERATION_CONTEXT, \
+    CIM_ERR_NOT_SUPPORTED, \
+    DEFAULT_NAMESPACE, \
+    MOFCompiler, MOFWBEMConnection
+from pywbem.cim_obj import NocaseDict
+
+
+__all__ = ['FakedWBEMConnection']
+
+# This is a server default if the open request MaxObjectCount is None
+_DEFAULT_MAX_OBJECT_COUNT = 101
+
+
+class FakedWBEMConnection(WBEMConnection):
+    """
+    A faked WBEMConnection class that mocks _imethodcall and _methodcall
+    WBEMConnection methods and creates equivalents for all of
+    the WBEMConnection methods that communicate with the server.
+    Because it only mocks _imethodcall and _methodcall all of the logic in
+    the various WBEMConneciton methods that prepare requests for the server
+    (ex. GetClass) is utilized and the parameters of the method calls are
+    tested/modified, etc.
```

Comment:
Again, these two methods are not documented. Instead of this paragraph, we could say that unless described otherwise, the operation methods behave exactly like their corresponding methods on WBEMConnection.

Reply:
Done